### PR TITLE
LGTM fixes

### PR DESCRIPTION
**Pull request by OfirMos**

### CODE REVIEW

File: src/execution_plan/ops/shared/print_functions.c [hunk 1]
```diff
@@ -17,7 +17,7 @@ static inline int _NodeToString(char *buf, uint buf_len, const char *alias, cons
 }
 
 int TraversalToString(const OpBase *op, char *buf, uint buf_len, AlgebraicExpression *ae) {
```
Comment (swilly22) [inline src/execution_plan/ops/shared/print_functions.c:19]: Shouldn't this function return `uint` ?

File: src/graph/entities/qg_edge.c [hunk 2]
```diff
@@ -63,10 +63,10 @@ void QGEdge_Reverse(QGEdge *e) {
 	QGNode_ConnectNode(e->src, e->dest, e);
 }
 
-int QGEdge_ToString(const QGEdge *e, char *buff, int buff_len) {
+int QGEdge_ToString(const QGEdge *e, char *buff, uint buff_len) {
```
Comment (swilly22) [inline src/graph/entities/qg_edge.c:66]: Shouldn't this function return `uint` ?

File: src/graph/entities/qg_edge.h [hunk 3]
```diff
@@ -43,7 +43,7 @@ int QGEdge_RelationID(const QGEdge *e, int idx);
 void QGEdge_Reverse(QGEdge *e);
 
 /* Gets a string representation of given edge. */
-int QGEdge_ToString(const QGEdge *e, char *buff, int buff_len);
+int QGEdge_ToString(const QGEdge *e, char *buff, uint buff_len);
```
Comment (swilly22) [inline src/graph/entities/qg_edge.h:46]: Shouldn't this function return `uint` ?

**Review by swilly22 — CHANGES_REQUESTED** [review]
a few questions

File: src/execution_plan/ops/shared/print_functions.c (hunk 1 above)
Comment (OfirMos) [reply src/execution_plan/ops/shared/print_functions.c:19]: The porblem that this result is being later returned from ExpandIntoToString and this should be the signature of the func by redis:
typedef int (*fpToString)(const struct OpBase *, char *, uint);

File: src/execution_plan/ops/shared/print_functions.c [hunk 4]
```diff
@@ -7,23 +7,18 @@
...
 }
 
-int TraversalToString(const OpBase *op, char *buf, uint buf_len, AlgebraicExpression *ae) {
-	uint offset = 0;
-	if(!ae) {
-		offset += snprintf(buf, buf_len, "%s", op->name);
-		return offset;
-	}
+void TraversalToString(const OpBase *op, sds *buf, AlgebraicExpression *ae) {
+	if(!ae) *buf = sdscatprintf(*buf, "%s", op->name);
```
Comment (swilly22) [inline src/execution_plan/ops/shared/print_functions.c:19]: I'm not sure about this line, was it here previously? the `if` condition doesn't make sense

File: src/execution_plan/ops/op_value_hash_join.c [hunk 5]
```diff
@@ -169,30 +169,29 @@ void _cache_records(OpValueHashJoin *op) {
...
-	offset += snprintf(buff + offset, buff_len - offset, " = ");
+	*buff = sdscatprintf(*buff, " = ");
 
 	AR_EXP_ToString(op->rhs_exp, &exp_str);
-	offset += snprintf(buff + offset, buff_len - offset, "%s", exp_str);
+	*buff = sdscatprintf(*buff, "%s", exp_str);
 	rm_free(exp_str);
 
-	return offset;
+	return;
```
Comment (swilly22) [inline src/execution_plan/ops/op_value_hash_join.c:194]: no need for this `return`

File: src/execution_plan/execution_plan_debug.c [hunk 6]
```diff
@@ -31,11 +32,11 @@ void ExecutionPlan_Print(const ExecutionPlan *plan, RedisModuleCtx *ctx) {
 	ASSERT(plan && ctx);
 
 	int op_count = 0;   // Number of operations printed.
-	char buffer[1024];
+	sds buffer = sdsempty();
```
Comment (swilly22) [inline src/execution_plan/execution_plan_debug.c:35]: where are we freeing this `buffer` ?

**Review by swilly22 — CHANGES_REQUESTED** [review]
The code looks so much cleaner! nice work.

File: src/execution_plan/ops/shared/print_functions.c (hunk 4 above)
Comment (OfirMos) [reply src/execution_plan/ops/shared/print_functions.c:19]: Yes it was here

**Review by swilly22 — APPROVED** [review]

**Pull request merged**